### PR TITLE
Add genid and vmx and vmx_ssh support

### PR DESCRIPTION
**Pull request by xiaodwan**
1) Support genid checking for windows guests
2) Support vmx and vmx_ssh in convert_from_file

Signed-off-by: Xiaodai Wang <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.latest8.arch_x86_64.raw_f.NFS.rhv.it_vddk: PASS (1702.86 s)
(1/2) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.linux.input_mode.vmx.vmx_default.latest7.output_mode.rhev.rhv: PASS (892.86 s)
(2/2) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.linux.input_mode.vmx.vmx_default.latest7.output_mode.rhev.rhv_upload: PASS (1296.41 s)
 (1/4) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.linux.input_mode.vmx.vmx_ssh.latest8.output_mode.rhev.rhv: PASS (1042.59 s)                                  
 (2/4) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.linux.input_mode.vmx.vmx_ssh.latest8.output_mode.rhev.rhv_upload: PASS (1395.49 s)
(1/1) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.windows.input_mode.vmx.vmx_default.win2019.output_mode.libvirt: PASS (641.83 s)
(1/1) type_specific.io-github-autotest-libvirt.convert_from_file.positive_test.windows.input_mode.vmx.vmx_default.win10.output_mode.libvirt: PASS (686.91 s)

**Comment by xiaodwan** [comment]
#https://github.com/avocado-framework/avocado-vt/pull/2108

**Comment by xiaodwan** [comment]
@chunfuwen @chloerh please help to check this pr, thanks.

File: v2v/tests/src/v2v_options.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def run(test, params, env):
     params['vmcheck_flag'] = False
     checkpoint = params.get('checkpoint', '')
     error_flag = 'strict'
+    estimate_file = ''
```
Comment (chloerh) [inline v2v/tests/src/v2v_options.py]: This line seems unnecessary.

Comment (xiaodwan) [reply v2v/tests/src/v2v_options.py]: This line can eliminate an error 'undefine variant' in some situations.

File: provider/v2v_vmcheck_helper.py [hunk 2]
```diff
@@ -443,3 +450,74 @@ def check_graphics(self, param):
...
+                # genid will not be in vmxml
+                if re.search(r'genid', self.vmxml):
+                    self.log_err('Unexpected genid in xml:\n%s' % self.vmxml)
+                return
+
+            genid_str = _compose_genid(*genid_list)
+            logging.debug('genid string is %s' % genid_str)
+
+            if not re.search(genid_str, self.vmxml):
+                self.log_err('Not find genid or genid is incorrect')
```
Comment (chunfuwen) [inline provider/v2v_vmcheck_helper.py]: I assume it need fail ,rather than just log error information

Comment (xiaodwan) [reply provider/v2v_vmcheck_helper.py]: V2V can record all failed checkpoints, and prints them out at last. the log_error function can implement that.

**Pull request merged**